### PR TITLE
accounts db/refactor accounts db test - test_partial_clean

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

Add hot storage test for test_partial_clean.

Continued on #814

#### Summary of Changes
- Add hot storage test for test_partial_clean

Fixes #

### CODE REVIEW

**Comment by HaoranYi** [comment]
The actual change is in the last commit.
Please only review the last change if you have already reviewed https://github.com/anza-xyz/agave/pull/814

File: accounts-db/src/accounts_db.rs [hunk 1]
```diff
@@ -14491,8 +14488,12 @@ pub mod tests {
         let account4 = AccountSharedData::new(4, 0, AccountSharedData::default().owner());
 
         // Store accounts into slots 0 and 1
-        db.store_uncached(0, &[(&account_key1, &account1)]);
-        db.store_uncached(0, &[(&account_key2, &account1)]);
+        if db.accounts_file_provider == AccountsFileProvider::HotStorage {
+            db.store_uncached(0, &[(&account_key1, &account1), (&account_key2, &account1)]);
```
Comment (HaoranYi) [inline accounts-db/src/accounts_db.rs]: HotStorage only allows to store to the storage for a particular slot once. So we need to batch up all the stores for a particular slot.

**Comment by HaoranYi** [comment]
Rebase to master to pickup #814.

File: accounts-db/src/accounts_db.rs (hunk 1 above)
Comment (brooksprumo) [reply accounts-db/src/accounts_db.rs]: Can this change work for both TieredStorage and AppendVec?

File: accounts-db/src/accounts_db.rs [hunk 2]
```diff
@@ -14514,8 +14515,12 @@ pub mod tests {
...
 
         // store into slot 2
-        db.store_uncached(2, &[(&account_key2, &account3)]);
-        db.store_uncached(2, &[(&account_key1, &account3)]);
+        if db.accounts_file_provider == AccountsFileProvider::HotStorage {
+            db.store_uncached(2, &[(&account_key2, &account3), (&account_key1, &account3)]);
+        } else {
+            db.store_uncached(2, &[(&account_key2, &account3)]);
+            db.store_uncached(2, &[(&account_key1, &account3)]);
+        }
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs]: Same here

File: accounts-db/src/accounts_db.rs (hunk 1 above)
Comment (HaoranYi) [reply accounts-db/src/accounts_db.rs]: yes. updated.

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Pull request merged**